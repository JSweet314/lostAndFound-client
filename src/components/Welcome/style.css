.welcome-greeting {
  color: #fff;
  font-size: 5rem;
  margin: 5rem 0 2.5rem 0;
  text-shadow: 2px 2px 5px rgba(196, 126, 89, 0.5);
}

.welcome-title {
  text-align: center;
  margin: 2rem 0 0 0;
  color: orange;
}

.welcome-call-to-action,
.welcome-instructions {
  width: 75%;
  margin: auto;
  text-align: center;
}

.welcome-call-to-action {
  font-size: 5.5rem;
  color: orange;
}