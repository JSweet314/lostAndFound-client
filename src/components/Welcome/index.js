import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const Welcome = ({ username }) => {
  return (username ?
    <h2 className='welcome-greeting'>{`Welcome, ${username}`}</h2> :
    <div className='welcome-greeting'>
      <h2 className='welcome-title'>Lost & Found</h2>
      <p className='welcome-instructions'>
        Here to help your organization track reported lost and found 
        items. Be sure to view existing reports prior to reporting a new item.
      </p>
      <p className='welcome-call-to-action'>
        Please log In to Report/Search
      </p>
    </div>
  );
};

Welcome.propTypes = {
  username: PropTypes.string
};

export default Welcome; 