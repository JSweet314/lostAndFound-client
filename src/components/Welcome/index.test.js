import React from 'react';
import {shallow} from 'enzyme';
import Welcome from './index';

describe('Welcome', () => {
  it('should match a snapshot', () => {
    const wrapper = shallow(<Welcome />);
    expect(wrapper).toMatchSnapshot();
  });
});