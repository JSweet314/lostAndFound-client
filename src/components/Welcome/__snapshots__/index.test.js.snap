// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome
    username=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="welcome-title"
>
          Lost & Found
</h2>,
        <p
          className="welcome-instructions"
>
          Here to help you or your organization track reported lost and found items. Be sure to view existing reports prior to reporting a new item.
</p>,
        <p
          className="welcome-call-to-action"
>
          Please log In to Report/Search
</p>,
      ],
      "className": "welcome-greeting",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lost & Found",
          "className": "welcome-title",
        },
        "ref": null,
        "rendered": "Lost & Found",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Here to help you or your organization track reported lost and found items. Be sure to view existing reports prior to reporting a new item.",
          "className": "welcome-instructions",
        },
        "ref": null,
        "rendered": "Here to help you or your organization track reported lost and found items. Be sure to view existing reports prior to reporting a new item.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please log In to Report/Search",
          "className": "welcome-call-to-action",
        },
        "ref": null,
        "rendered": "Please log In to Report/Search",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="welcome-title"
>
            Lost & Found
</h2>,
          <p
            className="welcome-instructions"
>
            Here to help you or your organization track reported lost and found items. Be sure to view existing reports prior to reporting a new item.
</p>,
          <p
            className="welcome-call-to-action"
>
            Please log In to Report/Search
</p>,
        ],
        "className": "welcome-greeting",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lost & Found",
            "className": "welcome-title",
          },
          "ref": null,
          "rendered": "Lost & Found",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Here to help you or your organization track reported lost and found items. Be sure to view existing reports prior to reporting a new item.",
            "className": "welcome-instructions",
          },
          "ref": null,
          "rendered": "Here to help you or your organization track reported lost and found items. Be sure to view existing reports prior to reporting a new item.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please log In to Report/Search",
            "className": "welcome-call-to-action",
          },
          "ref": null,
          "rendered": "Please log In to Report/Search",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Welcome should match a snapshot with username provided 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome
    username="jon"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Welcome, jon",
      "className": "welcome-greeting",
    },
    "ref": null,
    "rendered": "Welcome, jon",
    "type": "h2",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Welcome, jon",
        "className": "welcome-greeting",
      },
      "ref": null,
      "rendered": "Welcome, jon",
      "type": "h2",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
